sqlmesh/core/snapshot/evaluator.py

```diff
@@ -277,17 +286,24 @@ def create(
                 snapshots_with_table_names[snapshot].add(table.name)
                 table_deployability[table.name] = is_deployable
                 tables_by_schema[d.schema_(table.db, catalog=table.catalog)].add(table.name)
+                gateway_per_schema[d.schema_(table.db, catalog=table.catalog)] = (
```

Comment:
this doesn't seem right, why is there a a gateway for each schema? isn't the gateway dependent on the model?

Reply:
yes, but this is used in `_get_data_objects` and `_create_schemas` to ensure that the appropriate engine adapter is used there too. so when the corresponding model has an explicit gateway set, mapping this to the schemas that are used in these schema-specific queries as well